### DISCUSSION TITLE
Is there a way to switch modes with a keyword prefix?

### DISCUSSION THREAD

**Discussion by amithm7** [davatorium/rofi#1770] [General]
For example, in FireFox browser, type `g<space>` will change address bar to google search bar.

Similarly, I would like to switch rofi modes with a prefix keywords, maybe `prefix:` or `prefix<space>`.

To say, switch between the following modes:
- application menu (default)
- emoji picker
- file finder
- etc...

Is this possible currently?

---

Feature request for this: gh-1771

**Comment by TonCherAmi** [comment]
Take a look at [combi](https://davatorium.github.io/rofi/1.7.5/rofi.1/#combi):

> When using the combi mode, a !bang can be used to filter the results by modes. All modes that match the bang as a prefix are included. For example, say you have specified -combi-modes run,window,windowcd. If your query begins with the bang !w, only results from the window and windowcd modes are shown, even if the rest of the input text would match results from run.

**Reply by amithm7** [reply]
that's a filter.

I need rofi in `run` mode by default and change mode to `window` with a keyword prefix, and other modes based on keyword.

**Reply by DaveDavenport** [reply]
Yes, its a filter that filters on what you want, effectively you switch between modes. Combine  require-input listview option with what @TonCherAmi  suggests is the closest you can get.
I currently see two options to quickly switch between modes:

* Combi mode merging run/window and select mode with bang support.
* Use keybindings to switch between modes.

**Comment by amithm7** [comment]
> https://github.com/davatorium/rofi/discussions/1770#discussioncomment-4555657

> Use keybindings to switch between modes.

I found this solution most appealing, currently:

```sh
rofi \
	-modi drun \
	-modi window \
	-show drun
```

Enable 2 modes with `-modi` option, show `drun` mode by default.

Switch / Cycle between them using keybindings <kbd>Ctrl</kbd>+<kbd>Tab</kbd> (on Linux, by default, no configuration I did).
